charms/worker/k8s/src/config/cluster.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Learn more at: https://juju.is/docs/sdk
+
+"""Cluster configuration options."""
+
+from typing import Optional
+
+import ops
+
+from charms.k8s.v0.k8sd_api_manager import (
+    DNSConfig,
+    GatewayConfig,
+    IngressConfig,
+    LoadBalancerConfig,
+    LocalStorageConfig,
+    MetricsServerConfig,
+    NetworkConfig,
+    UserFacingClusterConfig,
+)
+
+
+def assemble_cluster_config(
+    charm: ops.CharmBase,
+    cloud_provider: Optional[str],
+    current: Optional[UserFacingClusterConfig] = None,
+) -> UserFacingClusterConfig:
+    """Retrieve the cluster config from charm configuration and charm relations.
+
+    Returns:
+        UserFacingClusterConfig: The expected cluster configuration.
+    """
+    if not current:
+        assembled = UserFacingClusterConfig()
+    else:
+        assembled = current.model_copy(deep=True)
+
+    _assemble_local_storage(charm, assembled)
+    _assemble_dns(charm, assembled)
+    _assemble_gateway(charm, assembled)
+    _assemble_network(charm, assembled)
+    _assemble_ingress(charm, assembled)
+    _assemble_metrics_server(charm, assembled)
+    _assemble_load_balancer(charm, assembled)
+    assembled.cloud_provider = cloud_provider
+    return assembled
+
+
+def _assemble_local_storage(charm: ops.CharmBase, assembled: UserFacingClusterConfig):
+    if not (ls := assembled.local_storage):
+        ls = assembled.local_storage = LocalStorageConfig()
+    ls.enabled = bool(charm.config["local-storage-enabled"])
```

Comment:
Could you please put these values in the literals module? I know it might be a bit cumbersome now, but it will really benefit us  in the long run.